### ISSUE TITLE
Add option to sxbp_walk_figure() for walking whole lines at once

### ISSUE THREAD

**Issue by saxbophone** [saxbophone/libsaxbospiral#235]
Currently, this function walks each edge of the line one unit at a time, so an edge longer than 1 unit will have the callback called multiple times for all the coördinates along its length.

This is necessary for bitmap representations of the figure, and presents a convenience to prevent the intermediate coördinates of each edge from having to be calculated as a second step, however it is redundant for SVG files (only the vertices of the line are needed), and is presenting a great hindrance to the work in PR #224

**Comment by saxbophone** [comment]
Suggested new function prototype:

```c
void sxbp_walk_figure(
    const sxbp_figure_t* figure,
    size_t scale,
    bool plot_vertices_only, // if true, only vertices are plotted
                             // otherwise all 1-unit points are plotted
    bool( *plot_point_callback)(sxbp_co_ord_t location, void* callback_data),
    void* callback_data
);
```

**Comment by saxbophone** [comment]
Closed in #236
